src/Java.Interop.Tools.TypeNameMappings/Java.Interop.Tools.TypeNameMappings/JavaNativeTypeManager.cs

```diff
@@ -262,35 +262,8 @@ static string GetSpecialExportJniType (string typeName, ExportParameterKind expo
 		// Keep in sync with ToJniNameFromAttributes(TypeDefinition)
```

Comment:
This should likewise update `ToJniNameFromAttributes(TypeDefinition)`...though offhand I'm not entirely sure how that would work.

I can't think of a way to make that work, after ~10s of thought. Can you think of a way?

If allowing `TypeDefinition` to use `IJniNameProviderAttribute.Name` can't work, then this is fine. I just want to ensure that's not possible.

Reply:
Do you mean, implementing cecil based reflection won't work?

It's true that it's desirable that we don't reference those types even via reflection (also for consistency). I'll try if it can become better.

Reply:
Updated to not explicitly reference those names. But it is impossible to identify that any attribute that has Name from the first argument. Hopefully that assumption doesn't bite us much...

Reply:
> Do you mean, implementing cecil based reflection won't work?

Kinda/sorta, yes. The issue is that in Reflection, we're not using *just* reflection data. We're also *constructing instances of the custom attributes*, and invoking the `IJniNameProviderAttribute.Name` property on those instances. This "leaves" the realm of Cecil-supported reflection -- we're not querying Reflection data, we're using runtime-created object instances -- which is why I couldn't think of a way to make this work.

It might not be workable, which means that the `TypeDefinition`-based codepath will need to continue explicitly "knowing about" every possible custom attribute and acting upon them, which in turn could complicate future maintenance, if/when we add a new naming-related custom attribute...

Reply:
> Updated to not explicitly reference those names

I like that change. It's brilliant.